File: app/src/main/java/ru/voronezhtsev/weatherapp/domain/IWeatherLocalRepository.java
package ru.voronezhtsev.weatherapp.domain;

import io.reactivex.Completable;
import io.reactivex.Maybe;
import ru.voronezhtsev.weatherapp.models.domain.WeatherInfo;

public interface IWeatherLocalRepository {

    Completable save(WeatherInfo weatherInfo);

    Maybe<WeatherInfo> getWeather();
}

File: app/src/main/java/ru/voronezhtsev/weatherapp/domain/IWeatherLocalRepository.kt
package ru.voronezhtsev.weatherapp.domain

import io.reactivex.Completable
import io.reactivex.Maybe
import ru.voronezhtsev.weatherapp.models.domain.Weather

interface IWeatherLocalRepository {
    fun save(weather: Weather): Completable
    val weather: Maybe<Weather>
}